app/clients/cloudwatch/aws_cloudwatch.py

```diff
@@ -0,0 +1,93 @@
+import json
+import os
+import re
+import time
+
+from boto3 import client
+
+from app.clients.cloudwatch import CloudwatchClient
+from app.cloudfoundry_config import cloud_config
+
+
+class AwsCloudwatchClient(CloudwatchClient):
+    """
+    This client is responsible for retrieving sms delivery receipts from cloudwatch.
+    """
+
+    def init_app(self, current_app, *args, **kwargs):
+        self._client = client(
+            "logs",
+            region_name=cloud_config.sns_region,
+            aws_access_key_id=cloud_config.sns_access_key,
+            aws_secret_access_key=cloud_config.sns_secret_key
+        )
+        super(CloudwatchClient, self).__init__(*args, **kwargs)
+        self.current_app = current_app
+        self._valid_sender_regex = re.compile(r"^\+?\d{5,14}$")
+
+    @property
+    def name(self):
+        return 'cloudwatch'
+
+    def _get_all_logs(self, my_filter, log_group_name):
+
+        # Check all events in the last 30 minutes
+        now = round(time.time() * 1000)
+        beginning = now - 30 * 60 * 1000
+
+        next_token = None
+        all_log_events = []
+        while True:
+            if next_token:
+                response = self._client.filter_log_events(
+                    logGroupName=log_group_name,
+                    filterPattern=my_filter,
+                    nextToken=next_token,
+                    startTime=beginning,
+                    endTime=now
+                )
+            else:
+                response = self._client.filter_log_events(
+                    logGroupName=log_group_name,
+                    filterPattern=my_filter,
+                    startTime=beginning,
+                    endTime=now
+                )
+            log_events = response.get('events', [])
+            all_log_events.extend(log_events)
+            next_token = response.get('nextToken')
+            if not next_token:
+                break
+        return all_log_events
+
+    def check_sms(self, message_id, notification_id):
+        """
+        Go through the cloudwatch logs, filtering by message id.  Check the success logs first.  If we find
+        the message id there, we are done.  Otherwise check the failure logs.  If we don't find the message
+        in the success or failure logs, raise an exception.  This method is called on a five minute delay,
+        which is presumably enough time for the cloudwatch log to be populated.
+        """
+        # TODO presumably there is a better way to get the aws account number
+        account_number = os.getenv("SES_DOMAIN_ARN")
```

Comment:
This crummy way we are getting the account number will be addressed next sprint by:

https://github.com/GSA/notifications-api/issues/258

Reply:
the Env variable SES_DOMAIN_ARN won't be set in cloud.gov environments. You should add a method to retrieve the account ID to cloudfoundry_config.py that currently basis it off the same variable that we get the email address, and then gets updated to the proper way after 258 is done.